libcalico-go/lib/selector/parser/stringset.go

```diff
@@ -20,51 +20,36 @@ type StringSet []string
 
 // Contains returns true if a given string in current set
 func (ss StringSet) Contains(s string) bool {
-	if len(ss) == 0 {
-		// Empty set or nil.
-		return false
-	}
-	// There's at least one item, do a binary chop to find the correct
-	// entry.  [minIdx, maxIdx) defines the (half-open) search interval.
-	minIdx := 0
-	maxIdx := len(ss)
-	for minIdx < (maxIdx - 1) {
-		// Select the partition index. The loop condition ensures that
-		// minIdx < partitionIdx < maxIdx so we'll always shrink the
-		// search interval on each iteration.
-		partitionIdx := (minIdx + maxIdx) / 2
-		partition := ss[partitionIdx]
-		if s < partition {
-			// target is strictly less than the partition, we can
-			// move maxIdx down.
-			maxIdx = partitionIdx
-		} else {
-			// Target is >= the partition, move minIdx up.
-			minIdx = partitionIdx
-		}
+	// Sever to the binary search impl in the stdlib.  Note: it returns
```

Comment:
Once I touched this code, I  realised it was a dupe of the stdlib version, might as well use that.